src/sentry/debug_files/artifact_bundle_indexing.py

```diff
@@ -0,0 +1,271 @@
+from __future__ import annotations
+
+import hashlib
+import logging
+from dataclasses import dataclass
+from datetime import datetime
+from typing import Any, Dict, List, NamedTuple, Optional, Tuple, TypeVar
+
+import sentry_sdk
+from django.db import router
+from django.utils import timezone
+
+from sentry.locks import locks
+from sentry.models.artifactbundle import (
+    NULL_STRING,
+    ArtifactBundle,
+    ArtifactBundleArchive,
+    ArtifactBundleFlatFileIndex,
+    ArtifactBundleIndexingState,
+    FlatFileIndexState,
+)
+from sentry.utils import json, metrics
+from sentry.utils.db import atomic_transaction
+from sentry.utils.locking.lock import Lock
+from sentry.utils.retries import TimedRetryPolicy
+
+logger = logging.getLogger(__name__)
+
+
+# We want to keep the bundle as being indexed for 600 seconds = 10 minutes. We might need to revise this number and
+# optimize it based on the time taken to perform the indexing (on average).
+FLAT_FILE_INDEXING_CACHE_TIMEOUT = 600
+
+
+@sentry_sdk.tracing.trace
+def mark_bundle_for_flat_file_indexing(
+    artifact_bundle: ArtifactBundle,
+    project_ids: List[int],
+    release: str | None,
+    dist: str | None,
+):
+    identifiers = []
+
+    for project_id in project_ids:
+        if release:
+            identifiers.append(
+                FlatFileIdentifier(project_id, release=release, dist=dist or NULL_STRING)
+            )
+
+        identifiers.append(FlatFileIdentifier(project_id, release=NULL_STRING, dist=NULL_STRING))
```

Comment:
This happens unconditionally, even if a release and dist exist. I assume that we do this so we can also look up a file without release/dist even if it had them when indexing?

Reply:
Yes, one is the per-release url-only index, the other one is the per-project index containing only debug_ids